python/lsst/resources/s3.py

```diff
@@ -118,6 +125,25 @@ def _translate_client_error(err: ClientError) -> None:
 class S3ResourcePath(ResourcePath):
     """S3 URI resource path implementation class."""
 
+    use_threads: bool | None = None
+
+    @property
+    def _transfer_config(self) -> TransferConfig:
+        if self.use_threads is None:
+            try:
+                # Use yaml.safe_load to be flexible with what is accepted.
+                self.use_threads = bool(yaml.safe_load(os.environ["LSST_S3_USE_THREADS"]))
```

Comment:
I'm a bit torn on using yaml here to parse a string when we are in control of the environment variable setting and can document it to be 0 or 1. Do we really have to support "no" and "false" as well?